apps/system/js/bluetooth_v2.js

```diff
@@ -0,0 +1,438 @@
+    if (this._adapter) {
+      this._debug('bluetooth state:' + this._adapter.state);
+      this._isEnabled = (this._adapter.state === 'enabled');
+      // send default bluetooth state so quick settings
+      // could be get updated
+      if (this._isEnabled) {
+        window.dispatchEvent(new CustomEvent('bluetooth-enabled'));
+      } else {
+        window.dispatchEvent(new CustomEvent('bluetooth-disabled'));
+      }
+
+      this._stateChangeHandler =
+        this._bluetoothAdapterHandler.bind(this);
+      this._adapter.addEventListener('attributechanged',
+        this._stateChangeHandler);
+    }
+
+    /*
+     * In file transfering case:
+     * since System Message can't be listened in two js files within a app,
+     * so we listen here but dispatch events to bluetooth_transfer.js
+     * while getting bluetooth file transfer start/complete system messages
+     */
+    navigator.mozSetMessageHandler('bluetooth-opp-transfer-start',
+      (transferInfo) => {
+        this._setProfileConnected(this.Profiles.OPP, true);
+        window.dispatchEvent(new CustomEvent('bluetooth-opp-transfer-start',
+          {transferInfo: transferInfo}));
+      }
+    );
+
+    navigator.mozSetMessageHandler('bluetooth-opp-transfer-complete',
+      (transferInfo) => {
+        this._setProfileConnected(this.Profiles.OPP, false);
+        window.dispatchEvent(new CustomEvent('bluetooth-opp-transfer-complete',
+          {transferInfo: transferInfo}));
+      }
+    );
+
+    // decouple bluetooth enable/disable function from other system part
+    window.addEventListener('request-enable-bluetooth',
+      this._enableHandler.bind(this));
+    window.addEventListener('request-disable-bluetooth',
+      this._disableHandler.bind(this));
+
+    Service.registerState('isEnabled', this);
+    // LazyLoader.load(['js/bluetooth_icon.js',
+    //                  'js/bluetooth_transfer_icon.js',
+    //                  'js/bluetooth_headphone_icon.js'], function() {
+    //   this.icon = new BluetoothIcon(this);
+    //   this.icon.start();
+    //   this.transferIcon = new BluetoothTransferIcon(this);
+    //   this.transferIcon.start();
+    //   this.headphoneIcon = new BluetoothHeadphoneIcon(this);
+    //   this.headphoneIcon.start();
+    // }.bind(this)).catch(function(err) {
+    //   console.error(err);
+    // });
+  },
+
+  /**
+   * Remove all EventListeners and adapters when default adapter
+   * is changed.
+   */
+  _removeEventListeners() {
+    this._bluetooth.removeEventListener('attributechanged',
+      this._defaultAdapterChangeHandler);
+    this._adapter.removeEventListener('attributechanged',
+      this._stateChangeHandler);
+    // remove getAdapter cached promise
+    if (this._cacheGetAdapterPromise) {
+      this._bluetooth.removeEventListener('attributechanged',
+        this._promiseAdapterChangeHandler);
+      this._cacheGetAdapterPromise = null;
+    }
+    this._adapter = null;
```

Comment:
The function here you done, not only remove event listener. You also clean up `this._adapter`. It also means that the function to reset default adapter which we got before. For readablility, It would be better to separate it out.